_posts/blog/2016-12-19-running-on-fedora-25.md

```diff
@@ -0,0 +1,335 @@
+parameters from a profile. We'll also make sure our preferred editor is
+installed (replace `vim` with your preferred editor):
+
+```bash
+sudo dnf install accumulo java-1.8.0-openjdk-devel tuned vim
+```
+
+It is possible to install only a specific Accumulo service, if desired, but it
+won't hurt to have extra packages installed and not used. For the single node
+setup, we'll need almost everything. For the multi-node setup, it might make
+more sense to be selective about which you choose to install on each node (for
+example, to only install `accumulo-tserver`).
+
+### Set up tuned
+
+(Optional) `tuned` can optimize your server settings, adjusting things like
+your `vm.swappiness`. To set up `tuned`, do:
+
+```bash
+sudo systemctl start tuned.service     # start service
+sudo tuned-adm profile network-latency # pick a good profile
+sudo tuned-adm active                  # verify the selected profile
+sudo systemctl enable tuned.service    # auto-start on reboots
+```
+
+### Set up ZooKeeper
+
+You'll need to set up ZooKeeper, regardless of whether you'll be running a
+single node or many. So, let's create it's configuration file (the defaults are
+fine):
+
+```bash
+sudo cp /etc/zookeeper/zoo_sample.cfg /etc/zookeeper/zoo.cfg
+```
+
+Now, let's start ZooKeeper (and set it to run on reboot):
+
+```bash
+sudo systemctl start zookeeper.service
+sudo systemctl enable zookeeper.service
+```
+
+Note that the default port for ZooKeeper is `2181`.
+
+## Running a single node
+
+### Configure Accumulo
+
+To run on a single node, you don't need to run HDFS. Accumulo can use the local
+filesystem as a volume instead. By default, it uses `/tmp/accumulo`. Let's
+change that to something which will survive a reboot:
+
+```bash
+sudo vim /etc/accumulo/accumulo-site.xml
+```
+
+Change the value of the `instance.volumes` property from `file:///tmp/accumulo`
+to `file:///var/tmp/accumulo` in the configuration file (or another preferred
+location).
+
+While you are editing the Accumulo configuration file, you should also change
+the default `instance.secret` from `DEFAULT` to something else. You can also
+change the credentials used by the `tracer` service now, too. If you use the
+`root` user, you'll have to set its password to the same one you'll use later
+when you initialize Accumulo. If you use another user name, you'll have to
+create that user later.
+
+### Configure Hadoop client
+
+Now, Hadoop's default local filesystem handler isn't very good at ensuring
+files are written to disk when services are stopped. So, let's use a better
+filesystem implementation for `file://` locations. Even though we're not
+running HDFS, the Hadoop client code used in Accumulo can still be configured
+by modifying Hadoop's configuration file:
+
+```bash
+sudo vim /etc/hadoop/core-site.xml
+```
+
+Add a new property:
+
+```xml
+  <property>
+    <name>fs.file.impl</name>
+    <value>org.apache.hadoop.fs.RawLocalFileSystem</value>
```

Comment:
Might want to consider including a caveat about this impl not supporting `fsync`